### ISSUE TITLE
Documentation eTraGo

### ISSUE THREAD

**Issue by lukasol** [openego/eTraGo#56]
- [x] technical implementation of read the docs to eTraGo
- [x] fill documentation with at least description of args

**Comment by wolfbunke** [comment]
short Info update:

- http://docs.readthedocs.io/en/latest/getting_started.html
- https://github.com/oemof/oemof/wiki/How-to-release

**Comment by ulfmueller** [comment]
I would say we do it in a similar structure as [oemof](https://github.com/oemof/oemof/tree/dev/doc). [ding0](https://github.com/openego/ding0/tree/dev/doc) looks also like that. 

As I see it we can already start documenting without having the connection to rtd (so this can be done in parallel).

I created a doc folder and three first rst files (6b5e96ccf4a93947a3974c37dac5a65091ee1c24):
- about
- getting_started
- usage -> where I would describe the usage in particular via the args

@lukasol if you like you can start some writing
@wolfbunke When I started thinking about this I forgot that in the end you took the task of setting up the documentation. I just started this a bit (so we can at least start some writing already). Please go ahead and built up the structure towards a read-the-docs documentation in the meantime.

**Comment by wolfbunke** [comment]
I was testing available names for eTraGo and eGo as well. Few are already gone like:
- http://ego.readthedocs.io/en/latest/

I saved for the beginning http://openego.readthedocs.io/en/latest/ now.

Further option could be 
- egotool
- or open-ego, ego-tool

For eTraGo I created  etrago.readthedocs.io

It is online available under:

- [ etrago.readthedocs.io](etrago.readthedocs.io)
-[ etrago.rtfd.io ](etrago.rtfd.io)

For the next step I connect the first Documentation with readthedocs

**Comment by wolfbunke** [comment]
The implementation of read-the-docs is started in:  60039a9

**Comment by ulfmueller** [comment]
cool! what a pity that ego is gone already... I think the openego domain you saved is a good plan B!
concerning your implementation, for now, I would get rid of the 'usage_details.rst'. I think the usage.rst is enough. At least I would not know how to differentiate these 2.

**Comment by wolfbunke** [comment]
ok few errors appears:

```
Error
Failed to import project: Failed to get code 
from 'https://github.com/openego/eTraGo.git' (git clone): 1 git clone
error output: No submodule mapping found in .gitmodules for path 'src/ego.io'
```
@lukasol @ulfmueller @BartelsJ  Why we have the src folder in the repository of eTraGo?

**Comment by ulfmueller** [comment]
I do not know what the src folder does, is and why it is there. Looking at the history, @BartelsJ created it and might know...

**Comment by wolfbunke** [comment]
ok, I was testing few things:

-  adding [requirements.readthedocs.txt](http://docs.readthedocs.io/en/latest/faq.html#can-i-document-a-python-package-that-is-not-at-the-root-of-my-repositoryl) on two folder levels
- changing [setup.py ](https://github.com/openego/eTraGo/blob/dev/setup.py#L20-L23)
- adding sphinx_rtd_theme to requirements.txt

I am stopping for now and take later a look into this bug

**Comment by lukasol** [comment]
I guess that's a merge error and was created by accident. I'm pretty sure you can delete the directory.

**Comment by wolfbunke** [comment]
ok, done

**Comment by wolfbunke** [comment]
ok, looks for me that it's working now:

-  http://etrago.readthedocs.io/en/latest/

**Comment by ulfmueller** [comment]
Nice! 
How to use eTraGo is mentioned twice...

**Comment by wolfbunke** [comment]
small bug in http://etrago.readthedocs.io/en/latest/py-modindex.html
bug maybe in https://github.com/openego/eTraGo/blob/dev/doc/conf.py#L155

**Comment by wolfbunke** [comment]
fixed by using: `sphinx-apidoc -o api/ ../etrago`

**Comment by wolfbunke** [comment]
add fist draft of a doc-string documentation for the etrago args in 561dcac

**Comment by lukasol** [comment]
I updated the docstrings in https://github.com/openego/eTraGo/commit/487bd09f22751d58498fbc5f10080402e5a45c8f

It's not complete yet, maybe one of you can do that...

In that context I also rearranged the args a bit and tried to give them some structure...

**Comment by s3pp** [comment]
I tried to figure out why readthedocs is not up-to-date with the code. rtd [will clone and install](http://docs.readthedocs.io/en/latest/builds.html) eTraGo in a virtual environment on their own server using setup.py and will then run the build process with 
> sphinx-build -b html . _build/html

This does not succeed atm, because some packages are not part of the setup.py and e.g. dependencies of snapshot_clustering_app.py are not met anymore by the recent version of ego.powerflow. In addition to prevent that scripts in contrast to modules are executed while building the documentation they need to have the name == main check in place. See [here](http://www.sphinx-doc.org/en/stable/man/sphinx-apidoc.html). That's how I understand it so far.

**Comment by lukasol** [comment]
I just completed the documentation of the args. Is there any progress on the technical front? If that's completed, I'd say we close this issue.

**Comment by lukasol** [comment]
@wolfbunke could you please update the general project info [here](https://github.com/openego/eTraGo/blob/dev/doc/conf.py#L88-L91) to the same structure we used in the scripts?

**Comment by lukasol** [comment]
I remove this ticket from Release 0.4.
@wolfbunke please assess if this is finished and can be closed or not.

**Comment by wolfbunke** [comment]
At the moment the args documentation can't be seen on http://etrago.readthedocs.io/en/latest/. I can do it on Monday.

**Comment by wolfbunke** [comment]
This issue is connected with #98 and solved.
